pkg/scheduling/v2/scheduler.go

```diff
@@ -755,22 +767,16 @@ func (s *Scheduler) tryAssign(
 					}
 
 					resultsCh <- &assignResults{
-						assigned: batchAssigned,
+						assigned:    batchAssigned,
+						rateLimited: batchRateLimited,
+						unassigned:  batchUnassigned,
 					}
 
 					return nil
 				})
 
 				if err != nil {
 					s.l.Error().Err(err).Msg("error assigning queue items")
-					return
```

Comment:
This was the early return statement which prevented us from sending the rest of the `assignResults` below.